Hotel.java

```diff
@@ -83,4 +84,23 @@ public boolean constraint_6(){
         }
         return numOfVip <= (0.1 * rooms.size());
     }
+
+    /**
+     * the name of service in hotel is unique
+     * @return
+     */
+    public boolean constraint_11(){
+        HashMap<Service, HotelService> list_of_Services = this.getServices();
+        boolean boolToReturn = true;
+        for (Service s1: list_of_Services.keySet()) {
+            for (Service s2: list_of_Services.keySet()) {
+                if(s1 != s2){
+                    if(!(s1.getServiceName().equals(s2.getServiceName()))){
+                        boolToReturn = false;
```

Comment:
you check if two service names are different then you set boolToReturn -> false.
it should be - if two service names are equal then u set boolToReturn -> false..
am i wrong?

Reply:
you right.
delete the "!"

Reply:
you are in love with the sign !